src/cultionet/callbacks.py

```diff
@@ -0,0 +1,163 @@
+import typing as T
+import filelock
+from pathlib import Path
+
+import geowombat as gw
+import rasterio as rio
+from rasterio.windows import Window
+import torch
+from pytorch_lightning.callbacks import BasePredictionWriter
+from torch_geometric.data import Data
+
+from .data.const import SCALE_FACTOR
+from .utils.reshape import ModelOutputs
+
+
+class LightningGTiffWriter(BasePredictionWriter):
+    def __init__(
+        self,
+        reference_image: Path,
+        out_path: Path,
+        num_classes: int,
+        ref_res: float,
+        resampling,
+        compression: str,
+        write_interval: str = 'batch'
+    ):
+        super().__init__(write_interval)
+
+        self.reference_image = reference_image
+        self.out_path = out_path
+        self.out_path.parent.mkdir(parents=True, exist_ok=True)
+
+        with gw.config.update(ref_res=ref_res):
+            with gw.open(reference_image, resampling=resampling) as src:
+                profile = {
+                    'crs': src.crs,
+                    'transform': src.gw.transform,
+                    'height': src.gw.nrows,
+                    'width': src.gw.ncols,
+                    # distance (+1) + edge (+1) + crop (+1) crop types (+N)
+                    # `num_classes` includes background
+                    'count': 3 + num_classes - 1,
```

Comment:
so num_classes is 2 in the current training model?

Reply:
That's correct. This is reserved for multi-crop learning, but we don't have it setup yet.